src/macro/eval/evalMain.ml

```diff
@@ -436,8 +439,52 @@ let rec value_to_expr v p =
 				let args = List.map (fun v -> value_to_expr v p) (Array.to_list e.eargs) in
 				(ECall (epath, args), p)
 		end
-
-	| _ -> exc_string ("Cannot convert " ^ (value_string v) ^ " to expr")
+	| VInstance {ikind = IStringMap smap} ->
+		begin
+			let decl = ref [] in
```

Comment:
This is alright, but it would be more idiomatic to use `StringHashtbl.fold` to construct the list. We don't even have to worry about order here because it's undefined anyway.